server/hotrod/src/main/scala/org/infinispan/server/hotrod/HotRodServer.scala

```diff
@@ -247,7 +248,19 @@ class HotRodServer extends AbstractProtocolServer("HotRod") with Log {
       clientListenerRegistry.setEventMarshaller(Option(marshaller))
    }
 
-   override def stop: Unit = {
+   def addCacheStore(name: String, instance: Object): Unit = {
```

Comment:
Please let me explain a little bit more the idea behind it.

Cache Loaders and Writers are added to the Cache during 'PersistenceManagerImpl' startup. 'PersistenceManagerImpl' is tied to a cache instance and contains a reference to its `Configuration`. During the startup process it scans the configuration and instantiates Cache Loaders and Writers. 

In other words adding a CacheLoader/Writer to `DeployedCacheStoreFactory` doesn't mean it is created - it is only available for instantiation. It will be turned on on if it used in configuration. 

Given your doubts I think I defined `CacheStoreFactoryRegistry` at wrong scope. It shouldn't be defined at Cache level, but this should rather a global component. I think it would be a better fit then.

Does it sound reasonable?